sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/models/events/IncomingCallEvent.java

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.callingserver.models.events;
+
+import com.azure.communication.callingserver.implementation.converters.CommunicationIdentifierConverter;
+import com.azure.communication.callingserver.implementation.models.CommunicationIdentifierModel;
+import com.azure.communication.common.CommunicationIdentifier;
+import com.azure.core.annotation.Immutable;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.Map;
+
+/** The AddParticipantsSucceededEvent model. */
+@Immutable
+public final class IncomingCallEvent implements CallingServerBaseEvent {
```

Comment:
Incoming Call Event is from Event Grid - should we handle them in our SDK?

Reply:
I feel like this can be useful on the Contoso side since they would have to process Incoming events and Validation event anyways, so I have moved one step forward to let the eventParser supports them as well. Currently the eventParser could consume both event grid events and cloud events with one public interface.

Reply:
fixed